rotkehlchen/tasks/assets.py

```diff
@@ -383,11 +385,45 @@ def update_aave_v3_underlying_assets(chains_aggregator: 'ChainsAggregator') -> N
     with chains_aggregator.database.conn.write_ctx() as write_cursor:
         chains_aggregator.database.set_static_cache(  # remember last task ran
             write_cursor=write_cursor,
-            name=DBCacheStatic.LAST_AAVE_V3_ASSETS_UPDATE,
+            name=cache_key,
             value=ts_now(),
         )
 
 
+def update_aave_v3_underlying_assets(chains_aggregator: 'ChainsAggregator') -> None:
+    """Fetch the Aave V3 underlying assets and populate `underlying_tokens_list` in globaldb"""
+    _update_lending_protocol_underlying_assets(
+        chains_aggregator=chains_aggregator,
+        providers_info=(
+            (ChainID.ETHEREUM, AAVE_V3_DATA_PROVIDER_ETH),
+            (ChainID.ETHEREUM, AAVE_V3_DATA_PROVIDER_ETH_OLD),
+            (ChainID.ETHEREUM, LIDO_AAVE_V3_DATA_PROVIDER_ETH),
+            (ChainID.ETHEREUM, ETHERFI_AAVE_V3_DATA_PROVIDER_ETH),
+            (ChainID.OPTIMISM, AAVE_V3_DATA_PROVIDER_EVM),
+            (ChainID.POLYGON_POS, AAVE_V3_DATA_PROVIDER_EVM),
+            (ChainID.ARBITRUM_ONE, AAVE_V3_DATA_PROVIDER_EVM),
+            (ChainID.BASE, AAVE_V3_DATA_PROVIDER_BASE),
+            (ChainID.GNOSIS, AAVE_V3_DATA_PROVIDER_GNO),
+            (ChainID.SCROLL, AAVE_V3_DATA_PROVIDER_SCRL),
+        ),
+        protocol=CPT_AAVE_V3,
+        cache_key=DBCacheStatic.LAST_AAVE_V3_ASSETS_UPDATE,
+    )
+
+
+def update_spark_underlying_assets(chains_aggregator: 'ChainsAggregator') -> None:
+    """Fetch the Spark underlying assets and populate `underlying_tokens_list` in globaldb"""
+    _update_lending_protocol_underlying_assets(
+        chains_aggregator=chains_aggregator,
+        providers_info=(
```

Comment:
can you add a small test with faked data to check that this task works as expected?

Reply:
i might have done a bit too much but it's vcr-ed.